```
11:46:40 [ERROR] Failures: 
11:46:40 [ERROR]   ModelProvisioningTest.testRequiredNodesAndDedicatedClusterControllers:1406 Unexpected exception type thrown ==> expected: <java.lang.IllegalArgumentException> but was: <java.lang.IllegalStateException>
11:46:40 [ERROR]   ModelProvisioningTest.testRequiringMoreNodesThanAreAvailable:1384 Unexpected exception type thrown ==> expected: <java.lang.IllegalArgumentException> but was: <java.lang.IllegalStateException>
11:46:40 [ERROR]   QuotaValidatorTest.test_deploy_above_quota_budget:51 expected: <The resources used cost $1.63 but your quota is $1.25: Contact support to upgrade your plan.> but was: <Cannot perform this on unspecified resources>
11:46:40 [ERROR]   QuotaValidatorTest.test_deploy_above_quota_budget_in_dev:88 expected: <The resources used cost $0.16 but your quota is $0.01: Contact support to upgrade your plan.> but was: <Cannot perform this on unspecified resources>
11:46:40 [ERROR]   QuotaValidatorTest.test_deploy_above_quota_budget_in_publiccd:62 expected: <publiccd: The resources used cost $1.63 but your quota is $1.00: Contact support to upgrade your plan.> but was: <Cannot perform this on unspecified resources>
11:46:40 [ERROR]   QuotaValidatorTest.test_deploy_above_quota_clustersize:40 expected: <Clusters testCluster exceeded max cluster size of 10> but was: <Cannot perform this on unspecified resources>
11:46:40 [ERROR]   QuotaValidatorTest.test_deploy_max_resources_above_quota:73 expected: <publiccd: The resources used cost $1.63 but your quota is $1.25: Contact support to upgrade your plan.> but was: <Cannot perform this on unspecified resources>
11:46:40 [ERROR]   QuotaValidatorTest.test_deploy_with_negative_budget:108 expected: <The resources used cost $-.-- but your quota is $--.--: Please free up some capacity.> but was: <Cannot perform this on unspecified resources>
11:46:40 [ERROR]   ContainerModelBuilderTest.container_cluster_with_invalid_name_throws_exception_when_hosted:194 Unexpected exception type thrown ==> expected: <java.lang.IllegalArgumentException> but was: <java.lang.IllegalStateException>
11:46:40 [ERROR]   ContainerModelBuilderTest.two_clusters_with_clashing_cluster_names_throws_exception_when_hosted:224 Unexpected exception type thrown ==> expected: <java.lang.IllegalArgumentException> but was: <java.lang.IllegalStateException>
11:46:40 [ERROR] Errors: 
11:46:40 [ERROR]   ModelProvisioningTest.containerWithZooKeeperJoiningServers:2425 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.containerWithZooKeeperSuboptimalNodeCountDuringRetirement:2406 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.require_that_config_override_and_explicit_proton_tuning_and_resource_limits_have_precedence_over_default_node_resource_tuning:2574 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.require_that_proton_config_is_tuned_based_on_node_resources:2501 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.require_that_resources_can_be_partially_specified:2525 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testAllow2ContentGroupsDown:2386 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testClusterMembership:252 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testCombinedCluster:284 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testCombinedClusterWithJvmHeapSizeOverride:320 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testCombinedClusterWithJvmOptions:380 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testContainerOnly:1682 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testDedicatedClusterControllers:1029 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testExclusiveNodes:1446 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testGroupsOfSize1:886 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testIllegalGroupSize:715 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testImplicitLogserverContainer:1157->testContainerOnLogserverHost:2596 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testJvmOptions:1702 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testLogForwarderInAdminCluster:1128 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testLogForwarderNotInAdminCluster:1082 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testLogserverContainerWhenDedicatedLogserver:1057->testContainerOnLogserverHost:2596 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testMinRedundancyMetByGroups:1899 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testMinRedundancyMetWithinGroup:1928 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testMultipleCombinedClusters:416 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testNoNodeTagMeansTwoNodes:2043 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testNoNodeTagMeansTwoNodesNoContent:2062 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testNodeCountForContentGroup:186 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testNodesJdisc:122 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testNonCombinedCluster:352 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testRedundancy1:1957 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testRedundancy2DownscaledToOneNodeButOneRetired:1285 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testRedundancyWithGroupsAndThreeRetiredNodes:1256 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testRequestingRangesMax:1612 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testRequestingRangesMin:1582 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testRequestingSpecificNodeResources:1552 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testSeparateClusters:229 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testSlobroksAreSpreadOverAllContainerClusters:990 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testSlobroksClustersAreExpandedToIncludeRetiredNodes:936 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testSlobroksClustersAreExpandedToIncludeRetiredNodesWhenRetiredComesLast:961 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testSlobroksOnContainersIfNoContentClusters:738 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testStatefulProperty:2346 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testThatStandaloneSyntaxWithClusterControllerWorksOnHostedManuallyDeployed:1833 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testThatStandaloneSyntaxWorksOnHostedManuallyDeployed:1801 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testThatStandaloneSyntaxWorksOnHostedVespa:1743 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testUseArm64NodesForAdminCluster:1645 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testUsingGroupSizeNotGroups:677 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testUsingGroups:632 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testUsingNodesAndGroupCountAttributes:525 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testUsingNodesAndGroupCountAttributesAndGettingJustOneNode:1367 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testUsingNodesAndGroupCountAttributesAndGettingTooFewNodes:1181 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testUsingNodesAndGroupCountAttributesWithoutDedicatedClusterControllers:786 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testUsingNodesCountAttributesAndGettingJustOneNode:1472 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelProvisioningTest.testUsingNodesCountAttributesAndGettingTooFewNodes:1323 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ClusterInfoTest.bcp_deadline_is_passed_in_cluster_info:65->requestedCapacityIn:215 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ClusterInfoTest.default_bcp_with_multiple_instances:182->requestedCapacityIn:215 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ClusterInfoTest.specifying_bcp_without_explicit_groups:141->requestedCapacityIn:215 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ClusterInfoTest.specifying_only_default_deadline_is_possible:105->requestedCapacityIn:215 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   DedicatedAdminV4Test.testDedicatedLogserverInHostedVespa:206->createModel:240 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   DedicatedAdminV4Test.testLogForwarding:154->createModel:229->createModel:240 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   DedicatedAdminV4Test.testModelBuilding:71->createModel:229->createModel:240 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   DedicatedAdminV4Test.testThatThereAre2SlobroksPerContainerCluster:128->createModel:229->createModel:240 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   MetricsProxyContainerClusterTest.access_logging_hosted:128 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   MetricsProxyContainerClusterTest.all_nodes_are_included_in_metrics_nodes_config:113 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   MetricsProxyContainerTest.hosted_application_propagates_node_dimensions:115 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   MetricsProxyContainerTest.metrics_v2_handler_is_set_up_with_node_info_config:128 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   MetricsProxyContainerTest.one_metrics_proxy_container_is_added_to_every_node:36 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   MetricsProxyContainerTest.one_metrics_proxy_container_is_added_to_every_node_also_when_dedicated_CCC:55 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   QuotaValidatorTest.test_deploy_under_quota:30 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ContentClusterRemovalValidatorTest.testContentRemovalValidation:28 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ContentClusterRemovalValidatorTest.testOverridingContentRemovalValidation:42 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ContentTypeRemovalValidatorTest.testContentTypeRemovalValidation:27 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ContentTypeRemovalValidatorTest.testOverridingContentTypeRemovalValidation:44 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   GlobalDocumentChangeValidatorTest.testChangGlobalAttribute:18->testChangeGlobalAttribute:28 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   IndexingModeChangeValidatorTest.testChangingIndexModeFromIndexedToStreaming:47 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   IndexingModeChangeValidatorTest.testChangingIndexModeFromIndexedToStreamingWhenDisallowed:28 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   IndexingModeChangeValidatorTest.testChangingIndexModeFromStoreOnlyToIndexed:61 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   RedundancyChangeValidatorTest.testChangingRedundancyToOne:26 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   RedundancyChangeValidatorTest.testChangingRedundancyToOneWithValidationOverride:43 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   RedundancyIncreaseValidatorTest.testOverridingContentRemovalValidation:40 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   RedundancyIncreaseValidatorTest.testRedundancyIncreaseValidation:24 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ResourcesReductionValidatorTest.fail_when_reducing_multiple_resources_by_over_50_percent:48 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ResourcesReductionValidatorTest.fail_when_reduction_by_over_50_percent:33 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ResourcesReductionValidatorTest.overriding_resource_decrease:73 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ResourcesReductionValidatorTest.reduction_is_detected_when_going_from_unspecified_resources_container:81 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ResourcesReductionValidatorTest.reduction_is_detected_when_going_from_unspecified_resources_content:111 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ResourcesReductionValidatorTest.reduction_is_detected_when_going_to_unspecified_resources_container:97 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ResourcesReductionValidatorTest.reduction_is_detected_when_going_to_unspecified_resources_content:126 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ResourcesReductionValidatorTest.reorganizing_resources_is_allowed:67 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ResourcesReductionValidatorTest.small_resource_decrease_is_allowed:61 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ResourcesReductionValidatorTest.testOverridingSizeReductionValidation:185 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ResourcesReductionValidatorTest.testSizeReductionValidationMinimalDecreaseIsAllowed:177 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ResourcesReductionValidatorTest.testSizeReductionValidationSelfhosted:159 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ResourcesReductionValidatorTest.testSizeReductionValidationWithUnspecifiedResourcesHosted:143 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   RedundancyOnFirstDeploymentValidatorTest.testOverridingRedundancyOnFirstDeploymentValidation:41 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   RedundancyOnFirstDeploymentValidatorTest.testRedundancyOnFirstDeploymentValidation:27 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ImplicitIndexingClusterTest.existing_jdisc_is_used_as_indexing_cluster_when_multitenant:43->buildMultiTenantVespaModel:59 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ContainerModelBuilderTest.cloud_account_without_nodes_tag:597 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ContainerModelBuilderTest.cluster_with_zookeeper:750 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ContainerModelBuilderTest.load_balancers_can_be_set:261->verifyAllowedUrns:345 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ContainerModelBuilderTest.singlenode_servicespec_is_used_with_hosted_vespa:580 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   StatelessOnnxEvaluationTest.testStatelessOnnxModelEvaluationWithGpu:105 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelAmendingTestCase.testModelAmending:76 » IllegalState Cannot perform this on unspecified resources
11:46:40 [ERROR]   ModelAmendingTestCase.testModelAmendingWithDedicatedCC:123 » IllegalState Cannot perform this on unspecified resources
```

--- a/config-model/src/main/java/com/yahoo/config/model/provision/InMemoryProvisioner.java
+++ b/config-model/src/main/java/com/yahoo/config/model/provision/InMemoryProvisioner.java
@@ -233,7 +233,7 @@ public class InMemoryProvisioner implements HostProvisioner {
 
     // Minimal capacity policies
     private NodeResources decideResources(NodeResources resources) {
-        return resources.withUnspecifiedNumbersFrom(resources);
+        return resources.withUnspecifiedNumbersFrom(defaultNodeResources);
     }
 
     private List<HostSpec> allocateHostGroup(ClusterSpec clusterGroup, NodeResources requestedResourcesOrUnspecified,

--- a/config-model/src/main/java/com/yahoo/config/model/provision/InMemoryProvisioner.java
+++ b/config-model/src/main/java/com/yahoo/config/model/provision/InMemoryProvisioner.java
@@ -233,6 +233,7 @@ public class InMemoryProvisioner implements HostProvisioner {
 
     // Minimal capacity policies
     private NodeResources decideResources(NodeResources resources) {
+        if (defaultNodeResources.isUnspecified()) return resources;
         return resources.withUnspecifiedNumbersFrom(defaultNodeResources);
     }
 

